torch_frame/data/dataset.py

```diff
@@ -116,9 +118,16 @@ def _get_mapper(self, col: str) -> TensorMapper:
         elif stype == torch_frame.categorical:
             index, _ = self.col_stats[col][StatType.COUNT]
             return CategoricalTensorMapper(index)
-        elif stype == torch_frame.multi_categorical:
-            index, _ = self.col_stats[col][StatType.COUNT]
-            return MultiCategoricalTensorMapper(index, sep=self.sep)
+        elif stype == torch_frame.multicategorical:
+            index, _ = self.col_stats[col][StatType.MULTI_COUNT]
+            if self.sep is None or col not in self.sep:
+                warnings.warn(
+                    "The separator is not specified for multicategorical"
+                    " column: {col}. ',' will be set as default separator.")
+                if self.sep is None:
+                    self.sep = {}
```

Comment:
We can initialize this as empty as part of `__init__`: `self.sep = sep or {}`.

Reply:
Also, I prefer making its type as either `str` or `Dict`, instead of `None`. If `str`, we use the shared separator across all columns.

Reply:
How about `Optional[Union[str, Dict[str, str]]]`? So it can take either `None`, `str` or `Dict[str, str]`.

Reply:
@weihua916 If we are to default it to `,` anyways, can we set it as `Union[str, Dict[str, str]]` and defaults to `,`?

Reply:
I think we can follow https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html, where the default is set to `sep = ","`
It is weird to have `sep=None` also mean `sep = ","`. I think `sep=None` should mean automatic inference of the separator, but that's tricky to implement...

Reply:
> If we are to default it to , anyways, can we set it as Union[str, Dict[str, str]] and defaults to ","?

Yes, that makes most sense to me. Thanks